### PR TITLE
Slide up panel for reaction list

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Created a component that can be used with a modal to display slide up panel.

This PR uses the slideUpPanel with the reaction lists

Feel free to ask any questions about the code.

Note: @saturninoabril if you have ideas on how that code can be tested please let me know.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12228

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

#### Device Information
This PR was tested on: [Device name(s), OS version(s)] 

#### Screenshots
image

screenshot_1540335324

### CODE REVIEW

File: app/components/slide_up_panel/slide_up_panel.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    Animated,
+    PanResponder,
+    Platform,
+    StyleSheet,
+    TouchableWithoutFeedback,
+    View,
+} from 'react-native';
+import DeviceInfo from 'react-native-device-info';
+
+const IS_IPHONE_X = DeviceInfo.getModel().includes('iPhone X');
```
Comment (saturninoabril) [inline app/components/slide_up_panel/slide_up_panel.js]: I'm seeing this in several places. Can we make this a utility function to find if Iphone is X?

File: app/components/slide_up_panel/slide_up_panel.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+            top: this.state.position,
+        };
+
+        return (
+            <View style={styles.viewport}>
+                <View
+                    ref={this.setBackdropRef}
+                    style={styles.backdrop}
+                    pointerEvents='box-only'
+                    onTouchEnd={() => {
```
Comment (saturninoabril) [inline app/components/slide_up_panel/slide_up_panel.js]: Can we add function for `onTouchEnd`?

File: app/components/slide_up_panel/slide_up_panel.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        justifyContent: 'center',
+    },
+    dragIndicator: {
+        backgroundColor: 'white',
+        height: 5,
+        width: 62.5,
+        opacity: 0.9,
+        borderRadius: 25,
+    },
+});
```
Comment (saturninoabril) [inline app/components/slide_up_panel/slide_up_panel.js]: Yeah, I have an idea how to do unit test for this component. Not about the animation but the typical unit testing where doing similar to other components. Probably, I'll add it up in the coming days.

**Comment by enahum** [comment]
@saturninoabril changes made

**Comment by esethna** [comment]
Thanks Elias!

A couple comments on the screens:

1. Looks like the header of the slide up panel is wider taller than what was speced:

image

2. The Android screens should have slightly different styling compared to iOS. Given the "floating bar" is not standard, and less rounded corners, see design:

image

**Comment by enahum** [comment]
@esethna the rounder corners and the drag indicator for Android I can remove, but for 1 the spec does not specify the height of the header

**Comment by enahum** [comment]
@esethna I found the height and I'll see if I can do something about it, but the defined height is only from the reaction list and it does not take into account the height needed for the other part of the component, anyway let me see what I can do

**Comment by enahum** [comment]
@esethna and I just want to mention the drag indicator is not standard anywhere as is not a standard component, but will remove anyway

File: app/constants/device.js [hunk 4]
```diff
@@ -16,5 +17,6 @@ export default {
     ...deviceTypes,
     DOCUMENTS_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Documents`,
     IMAGES_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Images`,
+    IS_IPHONE_X: DeviceInfo.getModel().includes('iPhone X'),
```
Comment (hmhealey) [inline app/constants/device.js]: Strictly speaking, these other constants aren't action types, so we might want to separate them at some point

File: test/setup.js [hunk 5]
```diff
@@ -17,6 +17,14 @@ jest.mock('NativeModules', () => {
 });
 jest.mock('NativeEventEmitter');
 
+jest.mock('react-native-device-info', () => {
+    return {
+        getVersion: () => '0.0.0',
+        getBuildNumber: () => '0',
+        getModel: () => 'iPhone X',
```
Comment (hmhealey) [inline test/setup.js]: Does it make sense for tests to default to iPhone X? I don't know what our usage numbers are between the two

**Comment by enahum** [comment]
@esethna PR is building you can try it once it finishes.

items 1, 2 and 3 should be covered

**Comment by enahum** [comment]
@esethna try one more time

**Pull request merged**